providers/implementations/keymgmt/dsa_kmgmt.c

```diff
@@ -190,16 +192,50 @@ static int dsa_exportkey(void *key, OSSL_CALLBACK *param_cb, void *cbarg)
     return ret;
 }
 
+/*
+ * Same function for domain parameters and for keys.
+ * "dpk" = "domain parameters & keys"
+ */
+static int dsa_get_dpk_params(void *key, OSSL_PARAM params[])
+{
+    DSA *dsa = key;
+    OSSL_PARAM *p;
+
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_BITS)) != NULL
+        && !OSSL_PARAM_set_int(p, DSA_bits(dsa)))
+        return 0;
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_SECURITY_BITS)) != NULL
+        && !OSSL_PARAM_set_int(p, DSA_security_bits(dsa)))
+        return 0;
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_SIZE)) != NULL
+        && !OSSL_PARAM_set_int(p, DSA_size(dsa)))
+        return 0;
```

Comment:
I disagree with this bit: `OSSL_PKEY_PARAM_SIZE` is not a key parameter at all, it is a parameter of an operation.

For the `EC` operations for example it is completely left to the whims of the provider developer what to return in this case: he could return the estimated output either of a signature operation or of a kexchange operation.

We should not push the handling of the `EVP_PKEY_size()` legacy ambiguity down the provider API, rather it should be core in charge of figuring out how to solve the ambiguity of the `EVP_PKEY_size()` definition.

I would expect a function in core that:
1. goes through the trouble of going through the operations supported by the `keymgmt` of a `pkey` using `OP_keymgmt_query_operation_name` (or the fallback of looking for signature, kexchange, etc. operations with the same name)
2. once the list of supported operations is available, go through it in the order we deem necessary to satisfy the legacy of `EVP_PKEY_size()` ambiguity